### DISCUSSION TITLE
How to make citar-open list "Library Files" before "Links".

### DISCUSSION THREAD

**Discussion by ghost** [emacs-citar/citar#654] [Q&A] [answered]
After select a entry by invoking `citar-open` it always list "Links" before "Library Files", is it possible to show "Library Files" before "Links" instead?
I couldn't find a user option for this, is there a function that I can override to achieve this behavior?

**Comment by bdarcus** [comment]
Not ATM.

EDIT: I'm actually not sure what determines the initial group order, as changing the order in the group function doesn't have an effect.

Note, however: at least on my setup, the group order will change depending on history. So if you select the last resource one time, and then run `citar-open` again and select the same reference, that group will be first.

@roshanshariff - thoughts on this?

**Reply by ghost** [reply]
> Note, however: at least on my setup, the group order will change depending on history. So if you select the last resource one time, and then run `citar-open` again and select the same reference, that group will be first.

My setup has same behavior, its not a big deal but in my workflow I seldom use links. I'll try hack vertico's sorting functions, maybe that's the way to do it.

**Reply by bdarcus** [reply]
Yeah; I get it. Different users have different priorities here, and it makes sense that more important stuff is at the top.

Perhaps it's something we'll figure out in time.

**Comment by ghost** [comment, marked as answer]
I think I got it:
```elisp
(vertico-multiform-mode)
(setq vertico-multiform-commands
      '((citar-open (vertico-sort-function . vertico-sort-alpha))))
```
Although it also affect the sorting of references, but I'm ok with that.

**Comment by roshanshariff** [comment]
Yeah, the ordering is determined by the ordering function. We currently don't have one, so your completion command might use frequency/recency. If we do add an ordering function, then it'll strictly use that ordering and no longer care about recency. So we'd have to make it optional.

**Reply by bdarcus** [reply]
It would only apply to select-resource though?

**Reply by roshanshariff** [reply]
Yes.

**Reply by bdarcus** [reply]
But that could be a problem if you select multiple references.

**Reply by roshanshariff** [reply]
Not necessarily. We could make the ordering function just change the order of the groups, but not the order of the items within each group.
